File: src/main/java/melya/controls/initialization/LabelInitializer.java
package melya.controls.initialization;

import com.quantumjockey.mvvmbase.controls.initialization.base.ControlInitializer;
import javafx.scene.control.Label;
import javafx.scene.control.Tooltip;

public class LabelInitializer extends ControlInitializer<Label> {

    /////////// Constructors ////////////////////////////////////////////////////////////////

    public LabelInitializer(Label label) {
        super(label);
    }

    /////////// Public Methods //////////////////////////////////////////////////////////////

    public void update(String text, String tooltip) {
        this.markupReference.setText(text);
        this.setTooltip(tooltip);
    }

    /////////// Private Methods /////////////////////////////////////////////////////////////

    private void setTooltip(String _tooltip) {
        if (_tooltip != null && !_tooltip.equals(""))
            this.markupReference.setTooltip(new Tooltip(_tooltip));
    }

}

File: src/main/scala/com/quantumjockey/melya/controls/initialization/LabelInitializer.scala
package com.quantumjockey.melya.controls.initialization

import com.quantumjockey.melya.controls.initialization.base.ControlInitializer
import javafx.scene.control.{Tooltip, Label}

class LabelInitializer(label: Label) extends ControlInitializer[Label](label) {

  // Public Methods

  def update(text: String, tooltip: String) {
    this.markupReference.setText(text)
    this.setTooltip(tooltip)
  }

  // Private Methods

  private def setTooltip(tooltip: String) {
    if (tooltip != null && !tooltip.equals(""))
      this.markupReference.setTooltip(new Tooltip(tooltip));
  }

}
